digital_comms/fixed_network/interventions.py

```diff
@@ -26,51 +26,36 @@ def get_all_assets_ranked(system, ranking_variable, asset_variable, technology,
         preference.
 
     """
-    if asset_variable == 'exchange':
-        system_level = system._exchanges
-    elif asset_variable == 'cabinet':
-        system_level = system._cabinets
-    elif asset_variable == 'distribution':
-        system_level = system._distributions
-    else:
-        raise ValueError('Did not recognise asset_variable')
-
     if ranking_variable == 'rollout_benefits':
-        assets = sorted(system_level,
+            assets = sorted(
+            (asset for asset in system_assets \
+            if getattr(asset, technology) != asset.total_prems),
             key=lambda item: item.rollout_benefits[technology], reverse=reverse_value)
 
     elif ranking_variable == 'rollout_costs':
-        assets = sorted(system_level,
+        assets = sorted(
+            (asset for asset in system_assets \
+            if getattr(asset, technology) != asset.total_prems),
             key=lambda item: item.rollout_costs[technology], reverse=reverse_value)
 
     elif ranking_variable == 'rollout_bcr':
-        assets = sorted(system_level,
+        assets = sorted(
+            (asset for asset in system_assets \
+            if getattr(asset, technology) != asset.total_prems),
             key=lambda item: item.rollout_bcr[technology], reverse=reverse_value)
 
     elif ranking_variable == 'total_potential_bcr':
-        assets = sorted(system_level,
+        assets = sorted(
+            (asset for asset in system_assets \
+            if getattr(asset, technology) != asset.total_prems),
             key=lambda item: item.total_potential_bcr[technology], reverse=reverse_value)
-
-    elif ranking_variable == 'max_rollout_costs':
-        #get distribution ranking by total upgrade costs
-        total_upgrade_costs = system.get_total_upgrade_costs(technology)
-        total_upgrade_costs = {k:(sum(j for j in v),) for k,v in total_upgrade_costs.items()}
-        total_upgrade_costs = sorted(total_upgrade_costs, key=lambda item: item, reverse=reverse_value)
-
-        #get_distributions
-        unranked_distributions = []
-        for asset in system_level:
-            unranked_distributions.append(asset)
-
-        #rank the distribution objects based on the total_upgrade_costs list
-        assets = [x for _, x in sorted(zip(total_upgrade_costs,unranked_distributions))]
     else:
         raise ValueError('Did not recognise ranking preference variable')
 
     return assets
 
 
-def decide_interventions(system, year, technology, policy, annual_budget,
+def decide_interventions(system_assets, year, technology, policy, annual_budget,
```

Comment:
As far as I can tell, the `adoption_cap` argument is not used, but this is computed in the `runner.py` file